numba_cuda/numba/cuda/tests/cudadrv/test_cuda_ndarray.py

```diff
@@ -456,6 +457,36 @@ def test_issue_8477(self):
         dev_array_from_host.copy_to_device(dev_array)
 
 
+class TestArrayMethod(CUDATestCase):
+    """Tests of the __array__() method via np.array"""
+
+    def test_np_array(self):
+        dev_array = cuda.to_device(np.asarray([1.0, 2.0, 3.0]))
+        host_array = np.array(dev_array)
+        np.testing.assert_equal(dev_array.copy_to_host(), host_array)
+
+    def test_np_array_dtype(self):
+        dtype = np.int32
+        dev_array = cuda.to_device(np.asarray([1.0, 2.0, 3.0]))
+        host_array = np.array(dev_array, dtype=dtype)
+        np.testing.assert_equal(
+            host_array,
+            dev_array.copy_to_host().astype(dtype)
+        )
+
+    @unittest.skipUnless(IS_NUMPY_2, "NumPy 1.x does not pass copy kwarg")
```

Comment:
Is this a sufficiently specific condition? I think this might only be newer releases like 2.1.0. If we can check both 2.0.0 and 2-latest, that would be good.

Reply:
I'm not sure I understand what you're saying, but I'll try to answer with some information: these unit tests will run if the NumPy version is 2.0 or greater. I skipped them when NumPy is 1.x because they fail, which results from it not passing the `copy` kwarg.

Reply:
Did you test both NumPy 2.0.0 and 2.1.0? I think there may be relevant changes in a minor, rather than major, version.

Reply:
If these fail with NumPy 2.0.0, we will need something more specific than `IS_NUMPY_2`. Perhaps something like `IS_NUMPY_2_1_OR_GREATER`.

Reply:
They pass with NumPy 2.0 and 2.1, and fail with 1.x.

Reply:
Okay, great. This is fine then. Approving!